fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/api/Persister.java

```diff
@@ -36,11 +37,14 @@
 
     /**
      * The persistence handling for the given operation using the provided session and mapping.
-     * @param session The session associated with the OCFL object
+     * 
+     * @param storageSession The persistent session in which persistence is occurring.
+     * @param objectSession The session associated with the OCFL object
      * @param operation The operation and associated data need to perform the operation.
      * @param mapping The mapping information needed to perform the persistence operation
      * @throws PersistentStorageException on failure
      */
-    void persist(final OCFLObjectSession session, final ResourceOperation operation, final FedoraOCFLMapping mapping)
+    void persist(PersistentStorageSession storageSession, final OCFLObjectSession objectSession,
```

Comment:
If I'm understanding this correctly, it appears that we are adding the dependency on the PersistentStorageSession in order to make it possible to access the ResourceHeaders of the HEAD state of the object.  Is that right?   If that is indeed the case,  wouldn't it be cleaner if we factored out the meat of the getHeaders() operation into a static function in OCFLSPersistentStorageUtils and have the persist method call that static helper rather than encumbering the Persister with PersistentStorageSession.

Reply:
Good call, it should be resolved now